metagraph/src/graph/alignment/aligner_helper.hpp

```diff
@@ -412,6 +561,157 @@ std::ostream& operator<<(std::ostream& out, const Alignment<NodeType> &alignment
     return out;
 }
 
+template <typename NodeType>
+class AlignmentSuffix {
+  public:
+    AlignmentSuffix(const Alignment<NodeType> &alignment, const DBGAlignerConfig &config,
+                    size_t k)
+          : alignment_(&alignment),
+            config_(&config),
+            begin_it_(alignment_->get_query().data()),
+            end_it_(alignment_->get_query_end()),
+            ref_begin_it_(alignment_->get_sequence().data()),
+            ref_end_it_(ref_begin_it_ + alignment_->get_sequence().size()),
+            cigar_begin_(alignment_->get_cigar(), alignment_->get_clipping()),
+            cigar_end_(alignment_->get_cigar(),
+                       alignment_->get_cigar().end()
+                          - static_cast<bool>(alignment_->get_end_clipping())),
+            cigar_it_(cigar_begin_),
+            score_(alignment_->get_score()),
+            trim_(0),
+            k_(k) {
+        assert(cigar_begin_ <= cigar_end_);
+        assert(cigar_it_ <= cigar_end_);
+    }
+
+    AlignmentSuffix& operator++();
+    AlignmentSuffix& operator--();
+
+    auto get_score() const { return score_; }
+    std::string_view get_query() const {
+        return std::string_view(begin_it_, end_it_ - begin_it_);
+    }
+
+    std::string_view get_sequence() const {
+        return std::string_view(ref_begin_it_, ref_end_it_ - ref_begin_it_);
+    }
+
+    Cigar::Operator get_front_op() const { return *cigar_it_; }
+    const CigarOpIterator& get_op_it() const {
+        assert(cigar_begin_ <= cigar_it_);
+        assert(cigar_it_ <= cigar_end_);
+        return cigar_it_;
+    }
+
+    bool eof() const { return cigar_it_ == cigar_end_; }
+    bool reof() const { return cigar_it_ == cigar_begin_; }
```

Comment:
if eof = "end of file", what is that file here?

Why not implement `pos()` and compare it to `size()`, or something like that?